### PR TITLE
minecraft example fixes + VR controller support

### PR DESCRIPTION
**Pull request by shpowley**
- desktop mode + keyboard: general player movement related fixes
- VR mode + controllers: added VR controller support for player movement

### CODE REVIEW

File: examples/minecraft/src/VRPlayerControl.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    controller_right = useXRControllerState('right')
+
+  let gamepad_info = {
+    connected_right: false,
+    connected_left: false,
+    thumbstick: null,
+    a_button: null
+  }
+
+  useEffect(() => {
```
Comment (bbohlender) [inline examples/minecraft/src/VRPlayerControl.jsx:17]: the evaluations made in this use effect better fit directly into the use frame imo, seems overlay complex

File: examples/minecraft/src/VRPlayerControl.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      playerJump()
+    }
+
+    if (playerMove && gamepad_info.connected_left) {
+      playerMove({
+        forward: gamepad_info.thumbstick.yAxis < 0,
+        backward: gamepad_info.thumbstick.yAxis > 0,
+        left: gamepad_info.thumbstick.xAxis < 0,
+        right: gamepad_info.thumbstick.xAxis > 0,
+        rotation: state.camera.rotation
```
Comment (bbohlender) [inline examples/minecraft/src/VRPlayerControl.jsx:46]: this should probably something like `state.camera.getWorldRotation(helperEuler)`

**Comment by bbohlender** [comment]
@shpowley very nice PR
I have a few nitpicks, which I can also adress myself, but in general looks good to me
will merge, once I get the chance to try out myself :)

File: examples/minecraft/src/VRPlayerControl.jsx (hunk 1 above)
Comment (shpowley) [reply examples/minecraft/src/VRPlayerControl.jsx:30]: jump check placed directly in useFrame. useEffect removed.

File: examples/minecraft/src/VRPlayerControl.jsx (hunk 2 above)
Comment (shpowley) [reply examples/minecraft/src/VRPlayerControl.jsx]: implemented suggestion. see latest `VRPlayerControl.jsx`. was this done correctly?
no noticeable difference compared to state.camera.rotation ..VR headset turn has almost identical rotation on movement.

**Comment by shpowley** [comment]
Implemented suggested fixes.
Added immersive VR right thumbstick player turning.
Also, added x-axis "wiggle tolerance" check to make forward/backward movement less "janky".

**Pull request merged**